packages/plugins/jwt/src/index.ts

```diff
@@ -82,8 +84,19 @@ export function useJWT(options: JwtPluginOptions): Plugin {
   return {
     async onRequestParse({ request, serverContext, url }) {
       const token = await getToken({ request, serverContext, url });
-      if (token != null) {
-        const signingKey = options.signingKey ?? (await fetchKey(jwksClient, token));
+      if (token) {
+        const decodedToken = decode(token, { complete: true });
+        if (!decodedToken?.header?.kid) {
+          throw unauthorizedError(`Failed to decode authentication token. Missing key id.`);
+        }
+
+        let signingKey: string;
+        if (jwksCache.has(decodedToken.header.kid)) {
+          signingKey = jwksCache.get(decodedToken.header.kid)!;
+        } else {
+          signingKey = await fetchKey(jwksClient, decodedToken.header.kid);
+          jwksCache.set(decodedToken.header.kid, signingKey);
+        }
```

Comment:
I personnaly prefere to avoid using mutable variables when possible :-) If it is ok for you too

```suggestion
        if (!jwksCache.has(decodedToken.header.kid)) {
          jwksCache.set(decodedToken.header.kid, await fetchKey(jwksClient, decodedToken.header.kid));
        }
        const signingKey = jwksCache.get(decodedToken.header.kid)!;
```

Reply:
Good call, much cleaner!